src/app/(nextjs_migration)/components/client/SignInButton.tsx

```diff
@@ -5,27 +5,30 @@
 "use client";
 
 import { signIn } from "next-auth/react";
+import { usePathname } from "next/navigation";
 import { useL10n } from "../../../hooks/l10n";
 
 export type Props = {
   autoSignIn?: boolean;
 };
 
-function initSignIn() {
-  void signIn("fxa", { callbackUrl: "/user/breaches" });
+function initSignIn(callbackUrl: string) {
```

Comment:
hm, how do we use this from e.g. src/app/(proper_react)/redesign/(authenticated)/user/welcome/page.tsx - does it need to be a prop like `autoSignIn`, so the caller can specify the right callback URL?

Reply:
It should be already working from `src/app/(proper_react)/redesign/(authenticated)/user/welcome/page.tsx`. We have the `SignInButton` set there: https://github.com/mozilla/blurts-server/blob/main/src/app/(proper_react)/redesign/(authenticated)/user/welcome/page.tsx#L18.

Reply:
Oh, I misunderstood where `pathname` was coming from, no this makes sense now thanks!